scout/adapter/mongo/omim.py

```diff
@@ -101,45 +110,41 @@ def omim_to_genes(self, omim_obj):
             gene_objs = [self.hgnc_gene_caption(hgnc_id) for hgnc_id in omim_obj.get("genes", [])]
         return gene_objs
 
-    def disease_term(self, disease_identifier):
-        """Return a disease term
-
-        Checks if the identifier is a disease number or a id
-
-        Args:
-            disease_identifier(str)
-
-        Returns:
-            disease_obj(dict)
-        """
+    def disease_term(
+        self,
+        disease_identifier: Union[str, int],
+        filter_project: Optional[dict] = DISEASE_FILTER_PROJECT,
+    ) -> dict:
+        """Return a disease term."""
         query = {}
         try:
             disease_identifier = int(disease_identifier)
             query["disease_nr"] = disease_identifier
         except ValueError:
             query["_id"] = disease_identifier
 
-        return self.disease_term_collection.find_one(query)
-
-    def disease_terms(self, hgnc_id=None):
-        """Return all disease terms that overlaps a gene
-
-            If no gene, return all disease terms
-
-        Args:
-            hgnc_id(int)
-
-        Returns:
-            iterable(dict): A list with all disease terms that match
-        """
+        if filter_project:
+            return self.disease_term_collection.find_one(query, filter_project)
+        else:
+            return self.disease_term_collection.find_one(query)
+
+    def disease_terms(
+        self,
+        hgnc_id: Optional[int] = None,
+        filter_project: Optional[dict] = DISEASE_FILTER_PROJECT,
+    ) -> list:
+        """Return all disease terms for a gene HGNC ID."""
         query = {}
         if hgnc_id:
             LOG.debug("Fetching all diseases for gene %s", hgnc_id)
             query["genes"] = hgnc_id
         else:
             LOG.info("Fetching all disease terms")
 
-        return list(self.disease_term_collection.find(query))
+        if filter_project:
+            return list(self.disease_term_collection.find(query, filter_project))
+        else:
+            return list(self.disease_term_collection.find(query))
```

Comment:
This looks like a bit of overkill to me. You have a default.  Is `find(query, None)` really a problem?
```suggestion
```

Reply:
Let's try!

Reply:
It works! 🥇